mlos_bench/mlos_bench/environments/script_env.py

```diff
@@ -78,17 +78,25 @@ def __init__(self,
         self._results_stdout_pattern: Optional[re.Pattern[str]] = \
             re.compile(results_stdout_pattern) if results_stdout_pattern else None
 
-    def _get_env_params(self) -> Dict[str, str]:
+    def _get_env_params(self, restrict: bool = True) -> Dict[str, str]:
         """
         Get the *shell* environment parameters to be passed to the script.
 
+        Parameters
+        ----------
+        restrict : bool
+            If True, only return the parameters that are in the `_shell_env_params`
+            list. If False, return all parameters in `_params` with some possible
+            conversions.
+
         Returns
         -------
         env_params : Dict[str, str]
             Parameters to pass as *shell* environment variables into the script.
             This is usually a subset of `_params` with some possible conversions.
         """
-        rename = {self._RE_INVALID.sub("_", key): key for key in self._shell_env_params}
+        input_params = self._shell_env_params if restrict else self._params.keys()
+        rename = {self._RE_INVALID.sub("_", key): key for key in input_params}
```

Comment:
A random thought on the `trial_id` vs `trial-id` issue mentioned above.
In this current code, we'd replace `trial-id` with `trial_id` and might have a conflict with the original `trial_id`.
I don't know how often that'd show up, but might be a good idea to issue a message about it at least.